plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/IndexMetadata.java

```diff
@@ -107,6 +156,27 @@ public DateTimeType(List<String> formats)
         {
             return formats;
         }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(formats.stream().sorted().collect(Collectors.toList()));
```

Comment:
why do we need the sorting? I'd avoid having to do such processing in `hashCode()`. If required we should cache this during construction